### PR TITLE
Point cloud 2d renderers, pt 1

### PR DESCRIPTION
**Pull request by nyalldawson**
Adds a bunch of the underlying API relating to 2d renderers for point clouds, and implements a "RGB" 2d renderer.

(no gui for now, sorry!)

Here's a screenshot of a RGB view of a point cloud (apologies for the poor quality cloud -- it's the only rgb one I have on hand immediately)

image

### CODE REVIEW

**Comment by nyalldawson** [comment]
@PeterPetrik next bit here :)

**Comment by nyalldawson** [comment]
Actually here's a better view:

image

File: src/core/pointcloud/qgspointcloudrgbrenderer.cpp [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+          red = *( short * )( ptr + i * recordSize + redOffset );
+          break;
+
+        case QgsPointCloudAttribute::Float:
+          red = *( float * )( ptr + i * recordSize + redOffset );
+          break;
+
+        case QgsPointCloudAttribute::Double:
+          red = *( double * )( ptr + i * recordSize + redOffset );
+          break;
```
Comment (PeterPetrik) [inline src/core/pointcloud/qgspointcloudrgbrenderer.cpp:104]: seeing this one in various places and probably more to come... wondering if we continue this way or some template function/helper function somewhere to have reusable code?

**Review by PeterPetrik — APPROVED** [review]

File: src/core/pointcloud/qgspointcloudrgbrenderer.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/core/pointcloud/qgspointcloudrgbrenderer.cpp:104]: I agree... I was going to try adding a few more renderers and see how much more duplicate code we end up with, and then try to remove it 😂

**Comment by stev-0** [comment]
On 2677d55d2e I'm getting nothing showing on any of the sample datasets or my own and the following in the debug log:
`totals: 0 nodes |  0  points |  0 ms`

which I guess isn't normal?

Running against PDAL | 2.2.0 (git-version: 7867d4)

**Comment by nyalldawson** [comment]
@stev-0 I'm guessing you're loading las/laz files directly? That's not fully supported just yet (it's a WIP) -- for now you need to convert them to binary entwine files before they can be viewed.

**Pull request merged**